pkg/controller/signup_test.go

```diff
@@ -328,6 +328,99 @@ func (s *TestSignupSuite) TestUpdateVerificationHandler() {
+		require.Equal(s.T(), "12268213044", storedUserSignup.Labels[crtapi.UserSignupUserPhoneHashLabelKey])
+	})
+	s.Run("init verification success phone number with spaces", func() {
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(rr)
+		data := []byte(`{"phone_number": "2 2 6 8 2 1 3 0 4 4", "country_code": "1"}`)
+		req, err := http.NewRequest(http.MethodPost, "/api/v1/signup/verification", bytes.NewBuffer(data))
+		require.NoError(s.T(), err)
+		ctx.Request = req
+		ctx.Set(context.SubKey, userID)
+
+		handler(ctx)
+
+		// Check the status code is what we expect.
+		assert.Equal(s.T(), http.StatusNoContent, rr.Code, "handler returned wrong status code")
+
+		// Check that the correct UserSignup is passed into the FakeSignupService for update
+		require.Equal(s.T(), userID, storedUserSignup.Name)
+		require.Equal(s.T(), "jsmith@redhat.com", storedUserSignup.Annotations[crtapi.UserSignupUserEmailAnnotationKey])
+		require.Equal(s.T(), "1", storedUserSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey])
+		require.Equal(s.T(), storedVerificationCode, storedUserSignup.Annotations[crtapi.UserSignupVerificationCodeAnnotationKey])
+		require.Equal(s.T(), verificationInitTimeStamp, storedUserSignup.Annotations[crtapi.UserSignupVerificationInitTimestampAnnotationKey])
+		require.Equal(s.T(), "12268213044", storedUserSignup.Labels[crtapi.UserSignupUserPhoneHashLabelKey])
+	})
+	s.Run("init verification success country code with parenthesis", func() {
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(rr)
+		data := []byte(`{"phone_number": "2268213044", "country_code": "(1)"}`)
+		req, err := http.NewRequest(http.MethodPost, "/api/v1/signup/verification", bytes.NewBuffer(data))
+		require.NoError(s.T(), err)
+		ctx.Request = req
+		ctx.Set(context.SubKey, userID)
+
+		handler(ctx)
+
+		// Check the status code is what we expect.
+		assert.Equal(s.T(), http.StatusNoContent, rr.Code, "handler returned wrong status code")
+
+		// Check that the correct UserSignup is passed into the FakeSignupService for update
+		require.Equal(s.T(), userID, storedUserSignup.Name)
+		require.Equal(s.T(), "jsmith@redhat.com", storedUserSignup.Annotations[crtapi.UserSignupUserEmailAnnotationKey])
+		require.Equal(s.T(), "1", storedUserSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey])
+		require.Equal(s.T(), storedVerificationCode, storedUserSignup.Annotations[crtapi.UserSignupVerificationCodeAnnotationKey])
+		require.Equal(s.T(), verificationInitTimeStamp, storedUserSignup.Annotations[crtapi.UserSignupVerificationInitTimestampAnnotationKey])
+		require.Equal(s.T(), "12268213044", storedUserSignup.Labels[crtapi.UserSignupUserPhoneHashLabelKey])
+	})
```

Comment:
All these four tests seem to be almost identical. You could just create a re-usable function like `verifyInitVerificationSuccess(rawPhoneNumber, expectedPhoneNumber string)`
and call it four times with `verifyInitVerificationSuccess("2 2 6 8 2 1 3 0 4 4", "2268213044")`, `verifyInitVerificationSuccess("(226) 821-3044", "2268213044")`, etc.

Reply:
@tinakurian to make the code even cleaner/more readable you could extract the actual phone normalization test to a custom assert function within the same test to cover all the cases.
Similar to what we do in other places. For example https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/idler/idler_controller_test.go#L341-L373
so some local function like `assertPhoneNormalized(rawPhoneNumber, expectedPhoneNumber)`

Reply:
@alexeykazakov i re-organized the test this way - is this okay? e8239d20e14faa8e1dc02e3dbe798aa6ff704e6a

Reply:
It's better. Thanks.